core/src/main/scala/akka/kafka/internal/TransactionalProducerStage.scala

```diff
@@ -228,25 +259,36 @@ private final class TransactionalProducerStageLogic[K, V, P](
   }
 
   override def onCompletionFailure(ex: Throwable): Unit = {
+    cancelTimer(commitSchedulerKey)
     abortTransaction("Stage failure")
-    batchOffsets.committingFailed()
-    super.onCompletionFailure(ex)
+
+    batchOffsets
+      .transactionAborted()
+      .onComplete {
+        case Success(_) =>
+          log.debug("Successfully sent source stage failure")
+          onCommitFailedAckCb.invoke(ex)
+        case Failure(commitEx) if NonFatal(commitEx) =>
```

Comment:
No need to check for `NonFatal`, futures wouldn't pass fatal exceptions.

Reply:
:+1: